### PR TITLE
Add the {*} matcher to Hash::extract()

### PR DESCRIPTION
**Pull request by markstory**
This matcher will match anything and is useful when you just want to traverse through data and you're not too picky.

I've also refactored the conditions to use a case as it is slightly more readable and uses fewer lines of code.

If people are happy with this, I'll forward port the changes to 3.x as well.

Refs #6447

### CODE REVIEW

File: lib/Cake/Test/Case/Utility/HashTest.php [hunk 1]
```diff
@@ -837,6 +837,28 @@ public function testExtractStringKey() {
 	}
 
 /**
+ * Test wildcard matcher
+ *
+ * @return void
+ */
+	public function testExtractWildcard() {
+		$data = array(
```
Comment (jrbasso) [inline lib/Cake/Test/Case/Utility/HashTest.php]: Your test case has only strings as key. The extract using `{s}` would cover that and will not justify the need of `{*}`. I would add different key types (ie, mixing numbers, strings, boolean, etc).

**Comment by jrbasso** [comment]
I like the change. :+1:  It is very handy in some cases.

File: lib/Cake/Test/Case/Utility/HashTest.php (hunk 1 above)
Comment (markstory) [inline lib/Cake/Test/Case/Utility/HashTest.php]: More cases added now.

**Comment by lorenzo** [comment]
Good Job!

**Comment by markstory** [comment]
I'll pick this onto 3.0 tonight.

**Pull request merged**